lib/idempotence/reservation/reservation.rb

```diff
@@ -107,17 +107,22 @@ def self.build
       class Reservation
         attr_reader :message
         attr_reader :idempotence_key
-        attr_reader :process_block
 
         def call(message, idempotence_key, &block)
           @message = message
           @idempotence_key = idempotence_key
 
-          yield message if @process_block
+          yield message if process_block
         end
 
-        def set_reserved(value)
-          @process_block = value
+        def process_block
+          @process_block ||= false
+        end
+        attr_writer :process_block
+
+        def set_reserved!
```

Comment:
Not sure about the `!` suffix - why is it there?

Reply:
Yeah, I think there is no need for that. I understand this happens in ruby core or activerecord just to give another meaning, like modifying the data structure or raising  an exception

Reply:
It's not common to use `set_` and `get_` in ruby but we do quite often in substitutes, e.g. `set_result` I think `set_reserved` is enough.